File: backend/src/Models/Booking.php
<?php

namespace Kivou\Models;

class Booking
{
    public int $id;
    public int $userId;
    public int $providerId;
    public string $serviceCategory;
    public ?string $serviceDescription;
    public string $scheduledAt; // 'Y-m-d H:i:s'
    public float $duration;
    public float $totalPrice;
    public string $status;
    public ?string $completedAt;

    public static function fromRow(array $r): self
    {
        $b = new self();
        $b->id = (int)$r['id'];
        $b->userId = (int)$r['user_id'];
        $b->providerId = (int)$r['provider_id'];
        $b->serviceCategory = $r['service_category'];
        $b->serviceDescription = $r['service_description'] ?? null;
        $b->scheduledAt = $r['scheduled_at'];
        $b->duration = (float)$r['duration'];
        $b->totalPrice = (float)$r['total_price'];
        $b->status = $r['status'];
        $b->completedAt = $r['completed_at'] ?? null;
        return $b;
    }

    public function json(): array
    {
        return [
            'id' => $this->id,
            'user_id' => $this->userId,
            'provider_id' => $this->providerId,
            'service_category' => $this->serviceCategory,
            'service_description' => $this->serviceDescription,
            'scheduled_at' => $this->scheduledAt,
            'duration' => $this->duration,
            'total_price' => $this->totalPrice,
            'status' => $this->status,
            'completed_at' => $this->completedAt,
        ];
    }
}

File: lib/models/booking.dart
/// Modèle représentant une réservation/commande
class Booking {
  final String id;
  final String userId;
  final String providerId;
  final String serviceCategory;
  final String serviceDescription;
  final DateTime scheduledAt;
  final double duration; // en heures
  final double totalPrice;
  final BookingStatus status;
  final String? customerNotes;
  final String? providerNotes;
  final DateTime createdAt;
  final DateTime? completedAt;
  final PaymentInfo? payment;

  Booking({
    required this.id,
    required this.userId,
    required this.providerId,
    required this.serviceCategory,
    required this.serviceDescription,
    required this.scheduledAt,
    required this.duration,
    required this.totalPrice,
    required this.status,
    this.customerNotes,
    this.providerNotes,
    required this.createdAt,
    this.completedAt,
    this.payment,
  });

  Booking copyWith({
    BookingStatus? status,
    String? customerNotes,
    String? providerNotes,
    DateTime? completedAt,
    PaymentInfo? payment,
  }) {
    return Booking(
      id: id,
      userId: userId,
      providerId: providerId,
      serviceCategory: serviceCategory,
      serviceDescription: serviceDescription,
      scheduledAt: scheduledAt,
      duration: duration,
      totalPrice: totalPrice,
      status: status ?? this.status,
      customerNotes: customerNotes ?? this.customerNotes,
      providerNotes: providerNotes ?? this.providerNotes,
      createdAt: createdAt,
      completedAt: completedAt ?? this.completedAt,
      payment: payment ?? this.payment,
    );
  }

  /// Durée formatée
  String get formattedDuration {
    if (duration == duration.floor()) {
      return '${duration.toInt()}h';
    } else {
      int hours = duration.floor();
      int minutes = ((duration - hours) * 60).round();
      return '${hours}h${minutes.toString().padLeft(2, '0')}';
    }
  }

  /// Prix formaté
  String get formattedPrice => '${totalPrice.toStringAsFixed(2)} €';
}

/// Statut d'une réservation
enum BookingStatus {
  pending('En attente'),
  confirmed('Confirmée'),
  inProgress('En cours'),
  completed('Terminée'),
  cancelled('Annulée'),
  refunded('Remboursée');

  const BookingStatus(this.label);
  final String label;
}

/// Informations de paiement
class PaymentInfo {
  final String method; // 'card', 'paypal', 'cash'
  final String? transactionId;
  final DateTime paidAt;
  final double amount;

  PaymentInfo({
    required this.method,
    this.transactionId,
    required this.paidAt,
    required this.amount,
  });
}
